techSpec.js

```diff
@@ -0,0 +1,209 @@
+/*
+2. Introduction
+
+a. Overview, Problem Description, Summary, or Abstract
+
+    Context: Cat API is an Express.js application run on Node.js runtime. 
+
+    Problem: The Cat API currently does not have access to a cat database that persists. 
+    The information is generated when the API server is started. 
+    For this reason, for example, the ids are different every time around which made writing test cases trickier. 
+
+    Suggested solution: Implement a SQLite database which allows information to be persisted. 
+
+b. Glossary  or Terminology
+
+    SQL - structured query language 
+    SQLite - a file-based database engine that uses SQL syntax
+    sqlite3 - node module enables connection to SQLite databases in memory or disk file 
+
+    sqlite3 methods:
+        db.all - allows us to access all the rows in the dataset - this might offer the best flexibility for now
+        db.each - executes code for each row in a dataset
+        db.get - executes code for the FIRST row - use when we know only one row will return. 
+
+d. Goals or Product and Technical Requirements
+
+    Product requirements: 
+        1. We should have a pre-populated SQLite database on the file system. 
+        2. Upon start-up of the Cat API Express server, it should establish a connection to the SQL database - or should it?  
+           Let's review:
+           The connection is required wherever we make a query via sqlite3 methods: all() get() and each()
+           The connection should be closed when we finish making the query and processing the result set. 
+           Therefore the connection does NOT need to be connected immediately upon server start-up.
+        3. The Cat API should be able to read from the SQLite database for GET all & GET by id requests and return objects. 
+        4. The Cat API should be able to write to the SQLite database for POST & PUT requests and return new/updated objects. 
+        4. The Cat API should be able to delete from the SQLite database for DEL requests. 
+        
+
+    Technical requirements: 
+        
+        1.
+        Create and populate 'cat' & 'pedigree breed' tables via sqlite in Terminal. 'CREATE TABLE' and so forth....
+        Remove list of cats and breed objects from catsRepo.js & breedsRepo.js respectively.   
+        Review if the repositories still need to be classes in this case? 
+        Note that id to have PRIMARY KEY constraint (needs to be unique)  
+         
+        2. 
+        Install sqlite3 node module into the root directory of cat api.
+        Import/require into repository files for use - Eg catsRepo.js is where all direct interaction with database happens. 
+        Repository methods will call a third party function that starts a database connection, queries the database and closes connection. 
```

Comment:
I mentioned this earlier before but the connection should always be open if possible. You could replace this part of the design with a function that obtains an already open connection.

Also the database connection needs to be shutdown when the ExpressJS server shuts down - you might need to research if there's an ExpressJS API that is called on shutdown.

Reply:
Don't know what you mean about the ExpressJS API being called. I have read some references to 'signals' though which I will explore more tomorrow, eg: 

process.on('SIGINT', () => {
    db.close();
    server.close();
});

Reply:
Yep, this is the right direction - great research! This all seems to be covered in another comment so I'll resolve this for now.